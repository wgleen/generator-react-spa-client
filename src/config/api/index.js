export default {
  baseURL: process.env.API_BASE_URL,
  timeout: process.env.API_TIMEOUT,
  headers: {
    contentType: process.env.API_HEADERS_CONTENT_TYPE,
    XKey: process.env.API_HEADERS_X_KEY
  }
}
