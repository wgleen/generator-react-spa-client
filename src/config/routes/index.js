const routes = {
  root: {
    path: '/'
  }
}

export default routes
