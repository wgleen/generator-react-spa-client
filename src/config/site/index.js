export default {
  charset: 'UTF-8',
  viewport: 'width=device-width, initial-scale=1',
  name: 'Generator React SPA Client',
  title: 'Generator React SPA Client',
  description: 'Generator React SPA Client',
  /*
    index: tells bots to index the page;
    noindex: tells bots not to index the page;
    follow: tells bots to crawl links on the page, and that you also vouch for them;
    nofollow: tells bots not to crawl links on the page, and that no endorsement is implied.
  */
  robots: process.env.ROBOTS,
  // Open Graph & Facebook
  openGraph: {
    status: false, // change it to true to use, default is false
    type: 'website',
    url: '', // set the website url, not the facebook page url
    title: '',
    description: '',
    image: ''
  },
  twitter: {
    status: false, // change it to true to use, default is false
    card: 'summary_large_image',
    url: '', // set the website url, not the twitter url
    title: '',
    description: '',
    image: ''
  }
}
