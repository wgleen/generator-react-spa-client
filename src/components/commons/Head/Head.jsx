import React from 'react'
import PropTypes from 'prop-types'
import { Helmet } from 'react-helmet'
import config from '../../../config/index'

const Head = (props) => {
  const { title, description, children } = props
  const { site: siteConfig } = config

  const titleTemplate = `${title || siteConfig.title} - ${siteConfig.name}`

  return (
    <Helmet>
      <title>{titleTemplate}</title>
      <meta
        name="description"
        content={description || siteConfig.description}
      />

      {children}
    </Helmet>
  )
}

Head.defaultProps = {
  title: '',
  description: '',
  children: null
}

Head.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  children: PropTypes.node
}

export default Head
