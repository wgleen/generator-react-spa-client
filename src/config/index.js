import api from './api'
import site from './site'
import routes from './routes'

const config = {
  api,
  site,
  routes
}

export default config
