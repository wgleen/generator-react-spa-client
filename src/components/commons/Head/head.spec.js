import React from 'react'
import { mount } from 'enzyme'
import { Helmet } from 'react-helmet'
import config from '../../../config/index'
import Head from './Head'

describe('<Head />', () => {
  it('should render correctly', () => {
    const component = mount(<Head />)

    expect(component).toMatchSnapshot()

    component.unmount()
  })

  it('should render the children prop given as html tag', () => {
    const component = mount(
      <Head>
        <meta name="author" content="John Doe" />
      </Head>
    )

    const helmet = Helmet.peek()

    const metaDescription = helmet.metaTags.find(
      (meta) => meta.name === 'author'
    )

    expect(metaDescription.content).toEqual('John Doe')

    component.unmount()
  })

  describe('when title prop is defined', () => {
    const title = 'Page Title'

    it('should render a title element with content equal to title prop given and siteConfig.name', () => {
      const component = mount(<Head title={title} />)

      const helmet = Helmet.peek()

      expect(helmet.title).toEqual(`${title} - ${config.site.name}`)

      component.unmount()
    })
  })

  describe('when title prop is not defined', () => {
    const title = undefined

    it('should render a title element with content equal to siteConfig.title and siteConfig.name', () => {
      const component = mount(<Head title={title} />)

      const helmet = Helmet.peek()

      expect(helmet.title).toEqual(`${config.site.title} - ${config.site.name}`)

      component.unmount()
    })
  })

  describe('when description prop is defined', () => {
    const description = 'Page description'

    it('should render a meta element name prop equal to "description" and content prop equal to description prop given', () => {
      const component = mount(<Head description={description} />)

      const helmet = Helmet.peek()

      const metaDescription = helmet.metaTags.find(
        (meta) => meta.name === 'description'
      )

      expect(metaDescription.content).toEqual(description)

      component.unmount()
    })
  })

  describe('when description prop is not defined', () => {
    const description = undefined

    it('should render a meta element name prop equal to "description" and content prop equal to description prop given', () => {
      const component = mount(<Head description={description} />)

      const helmet = Helmet.peek()

      const metaDescription = helmet.metaTags.find(
        (meta) => meta.name === 'description'
      )

      expect(metaDescription.content).toEqual(config.site.description)

      component.unmount()
    })
  })
})
