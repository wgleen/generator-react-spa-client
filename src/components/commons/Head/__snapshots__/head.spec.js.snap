// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Head /> should render correctly 1`] = `
<Head
  description=""
  title=""
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <SideEffect(NullComponent)
      defer={true}
      encodeSpecialCharacters={true}
      meta={
        Array [
          Object {
            "content": "Generator React SPA Client",
            "name": "description",
          },
        ]
      }
      title="Generator React SPA Client - Generator React SPA Client"
      titleAttributes={Object {}}
    >
      <NullComponent
        defer={true}
        encodeSpecialCharacters={true}
        meta={
          Array [
            Object {
              "content": "Generator React SPA Client",
              "name": "description",
            },
          ]
        }
        title="Generator React SPA Client - Generator React SPA Client"
        titleAttributes={Object {}}
      />
    </SideEffect(NullComponent)>
  </HelmetWrapper>
</Head>
`;
